library/spdm_crypt_lib/libspdm_crypt_cert.c

```diff
@@ -1061,6 +1064,21 @@ bool libspdm_x509_common_certificate_check(const uint8_t *cert, size_t cert_size
     end_cert_from_len = 64;
     end_cert_to_len = 64;
 
+    /* 0. check certificate size */
```

Comment:
Isn't this redundant? `libspdm_verify_cert_chain_data` already calls `ibspdm_x509_get_cert_from_cert_chain` before calling `libspdm_x509_certificate_check` and `libspdm_x509_common_certificate_check`.

Reply:
Yes, you are right. Looks like all the callers called `libspdm_x509_get_cert_from_cert_chain`. I guess I was confused by `libspdm_x509_set_cert_certificate_check_ex` unit tests, will remove /* 0. check certificate size */ and the two new tests in `libspdm_test_crypt_spdm_x509_set_cert_certificate_check_ex` below.

BTW, in  `libspdm_x509_get_cert_from_cert_chain`, do you think if I should also add a length check?
`if (current_cert + current_cert_len > cert_chain + cert_chain_length)`

Reply:
@steven-bellock Would you help review the latest commit? Thanks!